.github/workflows/shared-build-and-test-stronghold-nodejs.yml

```diff
@@ -0,0 +1,146 @@
+              build: |
+                npm run build && strip dist/*.node
+              test: |
+                npm run test
+            - host: ubuntu-latest
+              target: x86_64-unknown-linux-musl
+              architecture: x64
+              setup: |
+                sudo apt-get install -y musl-tools
+              build: |
+                export NPM_BUILD_NAPI_ARGS=--target=x86_64-unknown-linux-musl
+                npm run build && strip dist/*.node
+              test-docker: |
+                docker build -f .github/Dockerfile.alpine -t tester .
+                docker run --rm -v $(pwd):/build -w /build/bindings/stronghold-nodejs tester sh -c "readelf -d dist/identity-stronghold-nodejs.linux-x64-musl.node && npm run test"
+            - host: macos-latest
+              target: aarch64-apple-darwin
+              build: |
+                export NPM_BUILD_NAPI_ARGS=--target=aarch64-apple-darwin
+                npm run build && strip -x dist/*.node
+              # we can't test this properly because the macos-latest runners don't have apple silicon / aarch64
+              test: |
+                echo "let's hope for the best"
+      name: stable - ${{ matrix.settings.target }} - node@16
+      runs-on: ${{ matrix.settings.host }}
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Setup node
+          uses: actions/setup-node@v2
+          with:
+            node-version: 16
+            check-latest: true
+            architecture: ${{ matrix.settings.architecture }}
+
+        - name: Setup Rust and cache
+          uses: './.github/actions/rust/rust-setup'
+          with:
+            os: ${{ matrix.settings.host }}
+            job: ${{ github.job }}
+            target: ${{ matrix.settings.target }}
+
+        - name: Download bindings/wasm artifacts
+          uses: actions/download-artifact@v2
+          with:
+            name: ${{ inputs.input-artifact-name }}
+            path: bindings/wasm
+
+        - name: Prepare Wasm bindings link
+          run: |
+            npm ci
+            npm link
+          working-directory: ./bindings/wasm
+
+        - name: Setup
+          if: ${{ matrix.settings.setup }}
+          run: ${{ matrix.settings.setup }}
+
+        - name: Install dependencies
+          # ignore peer dependencies, because in the case of publishing the identity-wasm peer dependency is not released yet
+          run: npm ci --ignore-scripts --legacy-peer-deps && npm link @iota/identity-wasm
+
+        - name: Build
+          run: ${{ matrix.settings.build }}
+
+        - name: List packages
+          run: ls -R .
+
+        - name: Test bindings
+          if: ${{ inputs.run-tests && matrix.settings.test }}
+          run: ${{ matrix.settings.test }}
+
+        - name: Test bindings in docker
+          if: ${{ inputs.run-tests && matrix.settings.test-docker }}
+          run: ${{ matrix.settings.test-docker }}
+          working-directory: ./
```

Comment:
Do we need two steps for these? Both run arbitrary code, maybe we could replace `test-docker` with `test` unless `working-directory: ./` does something magic I'm not seeing?

Reply:
the default working directory is `bindings/stronghold` which is overridden here to mount the whole project folder into the container to make the link between the packages work